packages/fxa-auth-server/lib/payments/capability.ts

```diff
@@ -392,14 +392,24 @@ export class CapabilityService {
     if (!overlaps.length) return [SubscriptionEligibilityResult.CREATE];
 
     // Users with IAP Offering overlaps should not be allowed to proceed
-    if (
-      overlaps.some(
-        (overlap) =>
-          overlap.type === 'offering' &&
-          iapProductIds.includes(overlap.offeringProductId)
-      )
-    )
-      return [SubscriptionEligibilityResult.BLOCKED_IAP];
+    let iapBlockedPlan: AbbrevPlan | undefined;
+
+    const iapRoadblock = overlaps.some((overlap) => {
+      if (
+        overlap.type === 'offering' &&
+        iapProductIds.includes(overlap.offeringProductId)
+      ) {
+        iapBlockedPlan = iapSubscribedPlans.find(
```

Comment:
Generally I find it hard to read when variables outside of function dotchains are reassigned within the dotchain.

Reply:
Changing the `some` to a `find` here would solve the issue for you.